### PR TITLE
[Feature] Admin Dashboard Login Page

### PR DESCRIPTION
**Pull request by teoh4770**
**Why open another pr?**

💡 From Aidan:

I updated main for better formatting. Pull from main and update your changes, following the same patterns and router implemented.

i.e update, imports / exports, naming schemes, and routes

Submit another PR after you make the changes. 🥳

</aside>

**⭐ What have I add upon on the updates?**

- update the router implementation, with the login page.
- update the import/export formatting with the updated paths and aliases setting.
- update the file naming schemes, with the newly added pages and providers.
 (adminPage.tsx --> admin.page.tsx)

### CODE REVIEW

File: src/pages/Placeholder/Placeholder.page.tsx
Comment (juancwu) [inline src/pages/Placeholder/Placeholder.page.tsx:1]: Why was this modified? It seems the same to me.

Comment (teoh4770) [reply src/pages/Placeholder/Placeholder.page.tsx:1]: oh, this one is modified by @aidantrabs , and i just include it

Comment (teoh4770) [reply src/pages/Placeholder/Placeholder.page.tsx:1]: should i do something with it?

Comment (teoh4770) [reply src/pages/Placeholder/Placeholder.page.tsx:1]: I probably miss this one when commit it

File: tsconfig.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
             "@components": ["./src/components/index"],
             "@pages": ["./src/pages/index"],
             "@services": ["./src/services/index"],
+            "@providers": ["./src/providers/index"],
             "@utils": ["./src/utils/index"],
```
Comment (juancwu) [inline tsconfig.json:30]:
```suggestion
            "@utils": ["./src/utils/index"]
```
json does not have ending "," for last entry

**Review by juancwu — CHANGES_REQUESTED** [review]

File: src/main.tsx [hunk 2]
```diff
@@ -3,9 +3,12 @@ import ReactDOM from "react-dom/client";
 import "./index.css";
 
 import { Router } from "@utils";
+import { AuthProvider } from "@/providers/AuthProvider";
```
Comment (juancwu) [inline src/main.tsx:6]:
```suggestion
import { AuthProvider } from "@providers";
```

**Review by juancwu — APPROVED** [review]

File: src/main.tsx (hunk 2 above)
Comment (teoh4770) [reply src/main.tsx:6]: good call

File: tsconfig.json (hunk 1 above)
Comment (teoh4770) [reply tsconfig.json:30]: json "does not" need to have ending "," for last entry?

Comment (teoh4770) [reply tsconfig.json:30]: if so, i gonna change for every other json in the folder as well

**Review by juancwu — APPROVED** [review]

**Comment by juancwu** [comment]
There is an issue with firebase when running the tests because it is not mocked. I will take a look into it and add the changes into this PR.

File: src/pages/Placeholder/Placeholder.page.tsx
Comment (juancwu) [reply src/pages/Placeholder/Placeholder.page.tsx:1]: I know it was by Aidan. It shouldn't say that it is a commit, since it should have been merged or rebased from main.

Comment (teoh4770) [reply src/pages/Placeholder/Placeholder.page.tsx:1]: Alright 👍

**Comment by juancwu** [comment]
There are missing test files for admin page and login page but since it is not facing any users, we will deal with it later.

In addition, we will have to properly setup mocking for firebase for the unit tests.
